telegram/ext/_dispatcher.py

```diff
@@ -631,6 +670,70 @@ def remove_handler(self, handler: Handler, group: int = DEFAULT_GROUP) -> None:
             if not self.handlers[group]:
                 del self.handlers[group]
 
+    def drop_chat_data(self, chat_id: int = None, all_empty_entries: bool = False) -> None:
+        """Used for deleting a key from the :attr:`chat_data`.
+
+        .. versionadded:: 14.0
+
+        Args:
+            chat_id (:obj:`int`, optional): The chat id to delete from the persistence. The entry
+                will be deleted even if it is not empty. Mutually exclusive with
+                ``all_empty_entries``.
+            all_empty_entries (:obj:`bool`, optional): If :obj:`True`, all entries which evaluates
+                to :obj:`False` in a boolean context in :attr:`chat_data` will be deleted.
+                Mutually exclusive with ``chat_id``.
```

Comment:
Can you add an example along the lines "if `bool(chat_data[chat_id]) is False`, then it will be dropped"?

Actually, are we sure that we want this parameter at all? It might just lead to confusion and I don't think that we'd be using this method internally at all.
If an advanced user wants to clear "unsed" chat/user data, they can easily call `drop_chat/user_data` in a loop on their own with their own definiton of "empty".

maybe @Poolitzer can give his 2cts as well …

Reply:
> Actually, are we sure that we want this parameter at all? It might just lead to confusion and I don't think that we'd be using this method internally at all. If an advanced user wants to clear "unsed" chat/user data, they can easily call `drop_chat/user_data` in a loop on their own with their own definiton of "empty".

True, thats what the OP wanted though. But wouldn't there be some overhead calling `dp.update_persistence()` on every iteration in the loop?

Reply:
you mean this?

> Add another method for purging all orphaned keys from the persistence file.

IISC that rather means "drop all entries from the persistence file that are no longer present in `dispatcher.chat/user_data`. tbh, I don't see a big need for it as long as we make sure that we don't produce any "orphaned keys" - by calling `persistence.drop_*` in `dispatcher.drop_*_data`.